src/main/scala/fastq.scala

```diff
@@ -269,17 +269,16 @@ case object fastq {
 
     def parseFastqPhred33: Iterator[Option[FASTQ]] =
       lines
-        .filterNot(_.isEmpty)
         .grouped(4)
         .map { quartet =>
 
-          if(quartet(3) != "+") None
-          else {
+          if( quartet(2) startsWith "+" ) {
```

Comment:
To avoid such indexing mistakes, I would rewrite this piece like this:

```scala
.map { 
  case Seq(id, sequence, "+", quality) => 
    for {
      i <- Id.parseFrom(id)
      s <- Sequence.fromStringsPhred33(sequence, quality)
    } yield FASTQ(i, s)
  case _ => None
}
```

This is also better because it handles the case of an incomplete FASTQ entry properly: if at the end of the file there is an entry with _missing_ quality line, the group will contain 3 lines, the check on the 3rd line will pass, but the `quartet(3)` call will fail with out of bound exception 😱